zenith_utils/src/vec_map.rs

```diff
@@ -104,6 +104,24 @@ impl<K: Ord, V> VecMap<K, V> {
             VecMap(self.0[split_idx..].to_vec()),
         )
     }
+
+    /// Move items from [`other`] to the end of [`self`], leaving [`other`] empty.
+    /// If any keys in [`other`] is less than or equal to any key in [`self`],
+    /// [`InvalidKey`] error will be returned and no mutation will occur.
+    pub fn extend(&mut self, other: &mut Self) -> Result<(), InvalidKey> {
```

Comment:
Should we call this `append`, for consistency with the rust std collections? The return value is different, though. There's also https://doc.rust-lang.org/std/iter/trait.Extend.html, but this isn't quite that either.

Reply:
I wanted to, but I already have an `append` method for adding a `key` and `value`. I called it `append` over something like `insert` because the key must be > every other key in the map. I could think about renaming that function and then calling this append maybe?